File: climbStairs.py
#!/usr/bin/env python3

# https://leetcode.com/problems/climbing-stairs/

# fibonocci basically
def climbStairs(n: int) -> int:
    if (n <= 3):
        return n 
    a = 2
    b = 3
    for i in range(3, n):
        tmp = b
        b = a + b
        a = tmp
    return b

assert climbStairs(0) == 0
assert climbStairs(1) == 1
assert climbStairs(2) == 2
assert climbStairs(3) == 3
assert climbStairs(4) == 5
assert climbStairs(5) == 8
assert climbStairs(6) == 13
assert climbStairs(7) == 21

File: climbStairs.js
#!/usr/bin/env node

// https://leetcode.com/problems/climbing-stairs/

// fibonocci basically
var climbStairs = function(n) {
    if (n <= 3) { return n } 
    var a = 2;
    var b = 3;
    var tmp;
    for (var i = 3; i <= n-1; i++) {
        tmp = b
        b = a + b;
        a = tmp;
    }
    return b;
};

import { expect } from "./expect.js"

expect(climbStairs(0), 0);
expect(climbStairs(1), 1);
expect(climbStairs(2), 2);
expect(climbStairs(3), 3);
expect(climbStairs(4), 5);
expect(climbStairs(5), 8);
expect(climbStairs(6), 13);
expect(climbStairs(7), 21);
